### DISCUSSION TITLE
关于私聊的问题

### DISCUSSION THREAD

**Discussion by star4277** [simple-robot/Discussions#128] [Q&A]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

请问一下，当有人加入群的时候，我该如何实现对该陌生人的私聊

**Comment by ForteScarlet** [comment]
监听 [成员增加事件](https://docs.simbot.forte.love/main/simbot-api/love.forte.simbot.event/-member-increase-event/index.html) (或者mirai组件中的具体类型 [MiraiMemberJoinEvent](https://docs.simbot.forte.love/components/mirai/simbot-component-mirai-core/love.forte.simbot.component.mirai.event/-mirai-member-join-event/index.html)) 并获取 `member` ，然后向 `member` 发送消息。

```java
@Listener
public void onMemberIncrease(MemberIncreaseEvent event) {
    event.getMember().send(...);
}
```

> **Warning**
> mirai中bot对**非好友**目标发送消息**并不“稳定”**，很有可能因为各种原因无法发送（比如对方设置了禁止发起临时会话等）

**Reply by star4277** [reply]
![image](https://user-images.githubusercontent.com/54941990/236371957-15895337-dcf0-4755-8a05-821da27cf785.png)
我测试的时候被发送人和发送人都是开启的临时会话的，但是无法正常发送

**Reply by ForteScarlet** [reply]
临时会话权限开启与否只是可能性之一，bot主动发送临时会话是不可靠的。考虑引导用户添加bot为好友后再发送消息。
